source/channel_bootstrap.c

```diff
@@ -577,16 +578,16 @@ static void s_attempt_connection(struct aws_task *task, void *arg, enum aws_task
 
     if (status != AWS_TASK_STATUS_RUN_READY) {
         s_connection_args_release(task_data->args);
-        goto cleanup_task;
+        goto task_cancelled;
```

Comment:
trivial, but after this jump it's going to LOGF something about aws_last_error(), but we don't have a last error set, so it will log a random value.

Either move the LOGFs around, or raise an error here so something reasonable prints out later

Reply:
Good catch.